### PR TITLE
treeio draft

### PR DESCRIPTION
**Pull request by GuangchuangYu**
A blog post describing `treeio`, see also <https://github.com/ropensci/onboarding/issues/179#issuecomment-373259952>.

### CODE REVIEW

**Comment by stefaniebutland** [comment]
Thank you @GuangchuangYu. I'll review and give feedback by end of Thurs Mar 29.

**Comment by stefaniebutland** [comment]
Hi @GuangchuangYu. Your post is scheduled for publication on Tues May 1st.

Here is the preview: https://deploy-preview-163--ropensci.netlify.com/blog/2018/03/27/treeio/
Please update the date in YAML and for images and let me know if it's a good time for me to review and give feedback.

Thank you!

**Comment by GuangchuangYu** [comment]
Thanks @stefaniebutland. The date has been updated to May 1st.

**Comment by GuangchuangYu** [comment]
why it is not published as schedule?

**Comment by stefaniebutland** [comment]
@GuangchuangYu I'm sorry, this is my fault and I should have let you know I got behind in the blog schedule. I need to get your post reviewed and I plan to have feedback to you by Tuesday.

**Comment by sckott** [comment]
hi @GuangchuangYu - I'll be taking a look at it today and reviewing ...

File: content/blog/2018-05-01-treeio.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    - software
+    - packages
+    - data-access
+    - phylogeny
+    - data-integration
+    - tidydata
+    - reproducibility
+---
+
+Phylogenetic trees are commonly used to present evolutionary relationships of species. Newick is the *de facto* format in phylogenetic for representing tree(s). Nexus format incorporates Newick tree text with related information organized into separated units known as blocks. For the R community, we have [ape](https://cran.r-project.org/package=ape) and [phylobase](https://cran.r-project.org/package=phylobase) packages to import trees from Newick and Nexus formats. However, analysis results (tree + analysis findings) from widely used software packages in this filed are not well supported. Some of them are extended from Newick and Nexus (*e.g.* [RevBayes](https://revbayes.github.io/intro.html) and [BEAST](http://beast2.org/) outputs), while some of the others are just log files (*e.g.* [r8s](http://ginger.ucdavis.edu/r8s) and [PAML](http://abacus.gene.ucl.ac.uk/software/paml.html) outputs). Parsing these output files is important for interpreting analysis finding.
```
[the remark this answers is not in the document]
Comment (GuangchuangYu) [reply content/blog/2018-05-01-treeio.md]: done

File: content/blog/2018-05-01-treeio.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+Information associated with taxon species/strains can be meta-data (*e.g.* isolation host, time and location, *etc.*), phenotypic or experimental data. These data as well as analysis finding may be further analyzed in the evolutionary context for downstream integrative and comparative studies. Unfortunately, taxon-associated data are mostly stored in separate files and often in inconsistent formats. To fill this gap, I developed the [treeio](https://bioconductor.org/packages/treeio/) for phylogenetic tree data integration.
```
[the remark this answers is not in the document]
Comment (GuangchuangYu) [reply content/blog/2018-05-01-treeio.md]: yes.

File: content/blog/2018-05-01-treeio.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+| read.jplace      | parsing jplace file including output of EPA and pplacer                   |
+| read.mrbayes     | parsing output of MrBayes                                                 |
+| read.newick      | parsing newick string, with ability to parse node label as support values |
+| read.nhx         | parsing NHX file including output of PHYLDOG and RevBayes                 |
+| read.paml\_rst   | parsing rst file (output of BaseML or CodeML)                             |
+| read.phylip      | parsing phylip file (phylip alignment + newick string)                    |
+| read.r8s         | parsing output of r8s                                                     |
+| read.raxml       | parsing output of RAxML                                                   |
+
+For example, user can parse [BEAST](http://beast2.org/) output using `read.beast` function. All the information inferred by by [BEAST](http://beast2.org/) will be stored in the output object.
```
[the remark this answers is not in the document]
Comment (GuangchuangYu) [reply content/blog/2018-05-01-treeio.md]: corrected

File: content/blog/2018-05-01-treeio.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    ## Rooted; includes branch lengths.
+    ## 
+    ## with the following features available:
+    ##  'height',   'height_0.95_HPD',  'height_median',    'height_range', 'length',
+    ##  'length_0.95_HPD',  'length_median',    'length_range', 'posterior',    'rate',
+    ##  'rate_0.95_HPD',    'rate_median',  'rate_range'.
+
+### Linking external data to phylogeny
+
+In addition to analysis findings that are associated with the tree as we showed above, there is a wide range of heterogeneous data, including phenotypic data, experimental data and clinical data *etc.*, that need to be integrated and linked to phylogeny. For example, in the study of viral evolution, tree nodes may associated with epidemiological information, such as location, age and subtype. To facilitate data integration, [treeio](https://bioconductor.org/packages/treeio/) provides `full_join` method to link external data to phylogeny as demonstrated below:
```
[the remark this answers is not in the document]
Comment (GuangchuangYu) [reply content/blog/2018-05-01-treeio.md]: corrected

File: content/blog/2018-05-01-treeio.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+```
+
+![hexbin scatter plot: correlation of dN/dS with substitution rate](/img/blog-images/2018-05-01-treeio/hexbin-scatterplot.png)
+
+With this feature, users are able to compare/integrate evolutionary inferences obtained from different software packages/models.
+
+Further readings
+----------------
+
+For more details of the importing trees with data, please refer to the [vignette](http://bioconductor.org/packages/devel/bioc/vignettes/treeio/inst/doc/Importer.html). In addition, [treeio](https://bioconductor.org/packages/treeio/) also supports exporing tree data to BEAST Nexus format, which allows software output format conversion and also enables combining external data with tree into a single file.
```
[the remark this answers is not in the document]
Comment (GuangchuangYu) [reply content/blog/2018-05-01-treeio.md]: yes, and corrected.

File: content/blog/2018-05-01-treeio.md [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+
+The following screenshot shows an example of exporting [CODEML](http://abacus.gene.ucl.ac.uk/software/paml.html) output to BEAST Nexus file. The tree was visualized using FigTree and colored by *d<sub>N</sub>*. For more details, please refer to the vignette, [exporting trees with data](http://bioconductor.org/packages/devel/bioc/vignettes/treeio/inst/doc/Exporter.html).
+
+![](https://raw.githubusercontent.com/GuangchuangYu/treeio/master/vignettes/figures/codeml.png)
+
+All the data parsed/merged by [treeio](https://bioconductor.org/packages/treeio/) can be converted to tidy data frame using the [tidytree](https://cran.r-project.org/package=tidytree) and can be used to visualize and annotate the tree using [ggtree](https://bioconductor.org/packages/ggtree/).
```
[the remark this answers is not in the document]
Comment (GuangchuangYu) [reply content/blog/2018-05-01-treeio.md]: corrected.

**Comment by GuangchuangYu** [comment]
thanks @sckott.

**Comment by sckott** [comment]
thanks again @GuangchuangYu - we'll post in about 2 hrs

**Comment by GuangchuangYu** [comment]
thanks @sckott @stefaniebutland .

**Pull request merged**